ecs-cli/modules/config/command_config.go

```diff
@@ -63,43 +65,55 @@ func NewCommandConfig(context *cli.Context, rdwr ReadWriter) (*CommandConfig, er
 		return nil, errors.Wrap(err, "Error loading config")
 	}
 
-	// Determine Launch Type
-	// The launch type flag overrides default launch type stored in the local config
-	if launchTypeFromFlag := RecursiveFlagSearch(context, flags.LaunchTypeFlag); launchTypeFromFlag != "" {
-		ecsConfig.DefaultLaunchType = launchTypeFromFlag
+	// Configuration passed in via flags take precedence over stored config
+	err = ecsConfig.applyFlags(context)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error reading flags")
 	}
 
-	if err = ValidateLaunchType(ecsConfig.DefaultLaunchType); err != nil {
+	// Instantiate AWS Session
+	svcSession, err := ecsConfig.ToAWSSession(context)
+	if err != nil {
 		return nil, err
 	}
 
-	// Determine cluster
-	// Order of cluster resolution:
-	//  1) Inline flag
-	//  2) Environment Variable
-	//  3) ECS Config
-	if clusterFromEnv := os.Getenv(flags.ClusterEnvVar); clusterFromEnv != "" {
-		ecsConfig.Cluster = clusterFromEnv
+	// Determine Cloudformation StackName
+	if ecsConfig.Version == iniConfigVersion {
+		ecsConfig.CFNStackName = ecsConfig.CFNStackNamePrefix + ecsConfig.Cluster
```

Comment:
Why did we add this? I don't think this check was there before right

Reply:
This was for backwards compatibility for the `configure` workflows that used the old config file format. /c @PettitWesley